overlord/snapstate/snapstate.go

```diff
@@ -1114,6 +1139,79 @@ func RevertToRevision(st *state.State, name string, rev snap.Revision, flags Fla
 	return doInstall(st, &snapst, snapsup)
 }
 
+// TransitionCore transitions from an old core snap name to a new core
+// snap name. It is used for the ubuntu-core -> core transition (that
+// is not just a rename because the two snaps have different snapIDs)
+//
+// Note that this function makes some assumptions like:
+// - no aliases setup for both snaps
+// - no data needs to be copied
+// - all interfaces are absolutely identical on both new and old
+// Do not use this as a general way to transition from snap A to snap B.
+func TransitionCore(st *state.State, oldName, newName string) ([]*state.TaskSet, error) {
+	var oldSnapst, newSnapst SnapState
+	err := Get(st, oldName, &oldSnapst)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+	if !oldSnapst.HasCurrent() {
+		return nil, fmt.Errorf("cannot transition snap %q: not installed", oldName)
+	}
+
+	// FIXME: what user ID to use for an automatic transition?
```

Comment:
This may be unauthenticated since we allow core (and pretty much everything else ATM) to be installed by anyone, so it seems right to not auth for something that was auto-triggered.